lua/toggleterm.lua

```diff
@@ -184,6 +184,53 @@ function M.exec(cmd, num, size, dir, direction, go_back, open)
   term:send(cmd, go_back)
 end
 
+--- @param selection_type string
+--- @param trim_spaces boolean
+function M.send_lines_to_terminal(selection_type, trim_spaces)
+    -- trim_spaces defines if we should trim the spaces from lines which are sent to the terminal
+    if trim_spaces == nil then
+      trim_spaces = true
+  end
+
+  vim.validate({
+    selection_type = { selection_type, "string", true },
+    trim_spaces = { trim_spaces, "boolean", true },
+  })
+
+  -- Window number from where we are calling the function (needed so we can get back to it automatically)
+  local current_window = vim.api.nvim_get_current_win()
+  -- Line texts - these will be sent over to the terminal one by one
+  local lines = {}
+  -- Beginning of the selection: line number, column number
+  local b_line, b_col
+
+  if selection_type == "visual" then
+      -- Get the start and the end of the visual selection
+      b_line, b_col = unpack(vim.fn.getpos("'<"),2,3)
```

Comment:
TODO: This can be easily converted to an inner function, where we can use motions as well